target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-deco-m5.dtsi

```diff
@@ -0,0 +1,378 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	chosen {
+		bootargs-append = " root=/dev/mtdblock16";
```

Comment:
I've not been able to figure out why OpenWrt isn't finding the rootfs partition, because even though the bootloader passes in `root=mtd:1:rootfs rootfstype=squashfs rootwait` and fails to find `mtd:1:rootfs`, it should fall back to finding the rootfs by partition name after the `firmware` partition is split?

Reply:
I suspect that it's something to do with the stock U-Boot version and it potentially not supporting partition splitting. We could split `firmware` into `1:hlos` and `1:rootfs` like what the OEM firmware does, though we'd lose out on flexible kernel & rootfs partitions.